### PR TITLE
DC Asset last hostname generator

### PR DESCRIPTION
**Pull request by mkurek**
- it's possible now to get (or reserve) next (currently) free hostname for `DataCenterAsset`
- it's based on `AssetLastHostname`, used before for `BackOfficeAsset`
- template of hostname is specified for `NetworkEnvironment` (prefix, postfix and counter length)

### CODE REVIEW

File: src/ralph/networks/tests/test_models.py [hunk 1]
```diff
@@ -260,3 +265,40 @@ def test_delete_network_shouldnt_delete_related_ip(self):
         net.delete()
         ip.refresh_from_db()
         self.assertTrue(ip)
+
+
+class NetworkEnvironmentTest(RalphTestCase):
```
Comment (ar4s) [inline src/ralph/networks/tests/test_models.py:268]: add some test touching overflow `template_counter_length`

File: src/ralph/assets/models/assets.py [hunk 2]
```diff
@@ -263,6 +258,14 @@ def increment_hostname(cls, prefix, postfix=''):
         else:
             return obj
 
+    @classmethod
+    def get_next_free_hostname(cls, prefix, postfix, fill=5):
+        try:
+            last_hostname = cls.objects.get(prefix=prefix, postfix=postfix)
+        except cls.DoesNotExist:
+            last_hostname = cls(prefix=prefix, postfix=postfix)
```
Comment (ghost) [inline src/ralph/assets/models/assets.py:266]: Is this equivalent to `get_or_create(prefix=prefix, postfix=postfix)`?

File: src/ralph/networks/models/networks.py [hunk 3]
```diff
@@ -45,14 +45,24 @@ class NetworkEnvironment(NamedMixin):
...
         help_text=_(
-            'E.g. h<200,299>.dc|h<400,499>.dc will produce: h200.dc '
-            'h201.dc ... h299.dc h400.dc h401.dc'
-        ),
+            'This value will be used as a postfix when generating new hostname '
+            'in this network environment. For example, when prefix is "s1", '
+            'postfix is ".mydc.net" and counter length is 4, following '
+            ' hostnames will be generated: s10000.mydc.net, s10001.mydc.net, ..'
+            ', s19999.mydc.net.'
+        )
```
Comment (ghost) [inline src/ralph/networks/models/networks.py:65]: :+1: for doc

File: src/ralph/assets/models/assets.py (hunk 2 above)
Comment (mkurek) [inline src/ralph/assets/models/assets.py:266]: no - there is only creating object, but there is no saving if `DoesNotExist` is raised - is should only be created when new hostname is issued (not fetched/checked what will be the next hostname)

**Comment by ghost** [comment]
:ok:

**Comment by ar4s** [comment]
:ok:
